component/mimir/rules/kubernetes/types.go

```diff
@@ -39,7 +39,8 @@ func (args *Arguments) UnmarshalRiver(f func(interface{}) error) error {
 		return fmt.Errorf("mimir_namespace_prefix must not be empty")
 	}
 
-	return nil
+	// We must explicitly Validate because HTTPClientConfig is squashed and it won't run otherwise
+	return args.HTTPClientConfig.Validate()
```

Comment:
DefaultArguments on line 22 needs to provide a default value for HTTPClientConfig (sorry, it's outside of the changed hunk so I can't comment on that line directly)